File: moneylog_spring/src/main/java/bitc/fullstack503/moneylog_spring/dto/MonthlySummaryDTO.java
package bitc.fullstack503.moneylog_spring.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
public class MonthlySummaryDTO {
  @JsonProperty("memberId")
  private String memberId;
  @JsonProperty("totalIncome")
  private int totalIncome;
  @JsonProperty("totalExpense")
  private int totalExpense;
  private List<DailySummaryDTO> dailySummary;

  public MonthlySummaryDTO(String memberId, int totalIncome, int totalExpense, List<DailySummaryDTO> dailySummary) {
    this.memberId = memberId;
    this.totalIncome = totalIncome;
    this.totalExpense = totalExpense;
    this.dailySummary = dailySummary;
  }
}

File: moneylog_app/app/src/main/java/bitc/example/app/dto/MonthlySummaryDTO.kt
package bitc.example.app.dto

import com.google.gson.annotations.SerializedName


data class MonthlySummaryDTO (
  @SerializedName("totalIncome")
  val totalIncome: Int,  // 총 수입
  @SerializedName("totalExpense")
  val totalExpense: Int,  // 총 지출
  @SerializedName("dailySummary")
  val dailySummary: List<DailySummaryDTO> // 날짜별 요약 리스트
)